lib/item.js

```diff
@@ -30,7 +30,10 @@ Item.prototype.set = function (paramsOrKey, value) {
   if (_.isString(paramsOrKey)) {
     this.attrs[paramsOrKey] = value;
   } else {
-    this.attrs = _.merge({}, this.attrs, paramsOrKey);
+    this.attrs = _.mergeWith({}, this.attrs, paramsOrKey, (originalParam, sourceParam) =>
+      // Lodash's merge and mergeWith ignore empty arrays by default.
+      // This customizer function allows empty arrays to be set.
+      (_.isArray(sourceParam) ? sourceParam : undefined));
```

Comment:
I think this also changes the merge behaviour when arrays are not empty. Is this ok?
```js
const x = {
  'a': [{ 'b': 2 }, { 'd': 4 }]
};
 
const y = {
  'a': [{ 'c': 3 }, { 'e': 5 }]
};

_.merge({}, x, y); // { a: [{ b: 2, c: 3 }, { d: 4, e: 5 }] }

_.mergeWith({}, x, y, customizer); // { a: [{ c: 3 }, { e: 5 }] }
```

Reply:
Oh, good catch. It would be best to preserve the rest of the original behaviour as much as possible. I'll modify the condition so that the fix is only applied to empty arrays.